test/Microsoft.AspNet.Mvc.FunctionalTests/JsonPatchTest.cs

```diff
@@ -141,5 +255,29 @@ public static IEnumerable<object[]> InvalidJsonPatchData
             var body = await response.Content.ReadAsStringAsync();
             Assert.Equal(errorMessage, body);
         }
+
+        [Fact]
+        public async Task JsonPatch_InvalidData_FormatterErrorInModelState_Failure()
+        {
+            // Arrange
+            var server = TestHelper.CreateServer(_app, SiteName, _configureServices);
+            var client = server.CreateClient();
+
+            var input = "{ \"op\": \"add\", \"path\": \"Customer/Orders/2\", " +
+               "\"value\": { \"OrderName\": \"Name2\" }}";
+            var request = new HttpRequestMessage
+            {
+                Content = new StringContent(input, Encoding.UTF8, "application/json-patch+json"),
+                Method = new HttpMethod("PATCH"),
+                RequestUri = new Uri("http://localhost/jsonpatch/JsonPatchWithModelState")
+            };
+
+            // Act
+            var response = await client.SendAsync(request);
+
+            // Assert
+            var body = await response.Content.ReadAsStringAsync();
+            Assert.Equal("{\"\":[\"The input was not valid.\"]}", body);
+        }
```

Comment:
There is already a test for multiple operations - `JsonPatch_MultipleValidOperations_Success`